src/main/java/org/apache/sling/feature/maven/mojos/IncludeArtifactMojo.java

```diff
@@ -0,0 +1,255 @@
+            throw new MojoExecutionException("includeArtifactClassifier is not specified. Check your configuration");
+        }
+
+        checkPreconditions();
+
+        final Map<String, Feature> featuresMap = ProjectHelper.getFeatures(this.project);
+        Feature found = null;
+        String key = null;
+        for (final Map.Entry<String, Feature> entry : featuresMap.entrySet()) {
+            if (includeArtifactClassifier.equals(entry.getValue().getId().getClassifier())) {
+                key = entry.getKey();
+                found = entry.getValue();
+                break;
+            }
+        }
+        final Artifact art = new Artifact(new ArtifactId(this.project.getGroupId(), this.project.getArtifactId(),
+                this.project.getVersion(), null, this.project.getArtifact().getType()));
+        File file = null;
+        if (found == null) {
+            found = new Feature(new ArtifactId(this.project.getGroupId(), this.project.getArtifactId(),
+                    this.project.getVersion(), includeArtifactClassifier, FeatureConstants.PACKAGING_FEATURE));
+
+            file = new File(this.getTmpDir(), "feature-" + this.includeArtifactClassifier + ".json");
+            key = file.getAbsolutePath();
+            ProjectHelper.getFeatures(this.project).put(key, found);
+            ProjectHelper.getAssembledFeatures(this.project).put(key, found);
+        }
+        getLog().debug("Found Feature: " + found + ", artifact: " + art);
+        includeArtifact(found, includeArtifactExtension, art);
+        getLog().debug("Feature Key: " + key + ", feature from key: " + ProjectHelper.getAssembledFeatures(this.project).get(key));
+        includeArtifact(ProjectHelper.getAssembledFeatures(this.project).get(key), includeArtifactExtension,
+                art.copy(art.getId()));
+        // Add Dependencies if configured so
+        for(String includeDependencyScope: includeDependenciesWithScope) {
+            List<Dependency> dependencies = project.getDependencies();
+            getLog().info("Project Dependencies: " + dependencies);
+            for(Dependency dependency: dependencies) {
+                if(includeDependencyScope.equals(dependency.getScope())) {
+                    getLog().info("Include Artifact: " + dependencies);
+                    ArtifactId id = new ArtifactId(
+                        dependency.getGroupId(), dependency.getArtifactId(), dependency.getVersion(), dependency.getClassifier(), dependency.getType()
+                    );
+                    Artifact artifact = new Artifact(id);
+                    if(bundlesStartOrder >= 0) {
+                        artifact.setStartOrder(bundlesStartOrder);
+                    }
+                    found.getBundles().add(artifact);
+                }
+            }
+        }
+
+        // Obtain any features from Source folder and add any Extensions to the target feature
```

Comment:
This functionality looks odd, why is it needed? You can specify an existing feature model via the classifier configuration. If you want to include things from several features build an aggregate first.

Reply:
My idea is to include repoinit or other FM extension into the FM that is created on the bundle module. For example if a bundle needs a service user then it can specify this in a repoinit FM file, add it to the source and then this goal will include it. The included FM is just a snippet provided by the module and the only goal is to embed it into the FM of the module.
Because there is no FM from the bundle I cannot aggregate first. 
Not sure about the Classifier Configuration? How would that be used.

Reply:
This goal is to add bundle's FM into a local repository to assemble them later into a FM system. It also allows to add repoinit configuration to the FM so that a bundle can provide its necessary setup w/o a Installation Hook